LayoutTests/fast/viewport/ios/constant-width-viewport-with-no-scale-after-changing-view-scale.html

```diff
@@ -0,0 +1,87 @@
+<!DOCTYPE html> <!-- webkit-test-runner [ useFlexibleViewport=true ShouldIgnoreMetaViewport=true contentMode=desktop ] -->
+<html>
+<head>
+    <meta name="viewport" content="">
+    <style>
+        #square {
+            position: absolute;
+            width: 10vw;
+            height: 10vh;
+            border: 2px solid black;
+        }
+
+        #output {
+            width: 100%;
+            height: 100%;
+            overflow: scroll;
+        }
+
+        body {
+            margin: 0;
+            width: 100%;
+            height: 100%;
+        }
+    </style>
+    <script src="../../../resources/ui-helper.js"></script>
+    <script>
+
+    const metaViewportWidth = 390;
+
+    const viewportWidthBaseline = innerWidth;
+    const viewportHeightBaseline = innerHeight;
+    const initialScale = viewportWidthBaseline / metaViewportWidth;
+
+    document.querySelector('meta').setAttribute("content","width=" + metaViewportWidth);
+
+    function valuesAreClose(num1, num2, maxDiff = 0.5) {
+        return Math.abs(num1 - num2) <= maxDiff;
+    }
+    
+    if (window.testRunner) {
+        testRunner.dumpAsText();
+        testRunner.waitUntilDone();
+    }
+
+    async function runTest() {
+        const appendOutput = message => {
+            output.appendChild(document.createTextNode(message));
+            output.appendChild(document.createElement("br"));
+        };
+
+        for (let targetScale of [0.5, 0.75, 1, 1.25, 1.5, 1.25, 1, 0.75, 0.5]) {
+            appendOutput(`setViewScale(${targetScale.toFixed(2)})`);
+
+            await UIHelper.setViewScale(targetScale);
+            await UIHelper.ensurePresentationUpdate();
+
+            const actualZoomScale = (await UIHelper.zoomScale());
+            const expectedZoomScale = targetScale * initialScale;
+
+            const expectedWindowWidth = metaViewportWidth / targetScale;
+            const expectedWindowHeight = viewportHeightBaseline / expectedZoomScale;
+
+            const expectedSquareWidth = expectedWindowWidth / 10;
+            const expectedSquareHeight = expectedWindowHeight / 10;
+
+            if (valuesAreClose(expectedWindowWidth, innerWidth) && valuesAreClose(expectedWindowHeight, innerHeight)) { appendOutput(`PASS:  window size scaled correctly`); }
+            else { appendOutput(`FAIL:  window size should be within 0.5 of [${expectedWindowWidth}, ${expectedWindowHeight}], but was [${innerWidth}, ${innerHeight}]`); }
```

Comment:
Nit - per WebKit style guidelines, this should probably be:

```suggestion
            if (valuesAreClose(expectedWindowWidth, innerWidth) && valuesAreClose(expectedWindowHeight, innerHeight))
                appendOutput(`PASS:  window size scaled correctly`);
            else
                appendOutput(`FAIL:  window size should be within 0.5 of [${expectedWindowWidth}, ${expectedWindowHeight}], but was [${innerWidth}, ${innerHeight}]`);
```

(Same for the other two `if/else` statements below)